descartes_light/core/include/descartes_light/types.h

```diff
@@ -12,8 +15,8 @@ using State = Eigen::Matrix<FloatType, Eigen::Dynamic, 1>;
 template <typename FloatType>
 struct StateSample
 {
-  StateSample(const State<FloatType>& state_, FloatType cost_) : state(state_), cost(cost_) {}
-  StateSample(const State<FloatType>& state_) : StateSample(state_, static_cast<FloatType>(0.0)) {}
+  StateSample(const Eigen::Ref<const State<FloatType>>& state_, FloatType cost_) : state(state_), cost(cost_) {}
+  StateSample(const Eigen::Ref<const State<FloatType>>& state_) : StateSample(state_, static_cast<FloatType>(0.0)) {}
```

Comment:
Same comment for using `Eigen::Ref` in classes. It seems like this doesn't add any value because using `Ref` doesn't allow us to pass by value and `move` into the class member

Reply:
Yea, here it probably does not add value with move semantics, but before it would eliminate one copy if you passed a block, head, row, etc.